comm/src/test/scala/coop/rchain/p2p/EffectsTestInstances.scala

```diff
@@ -0,0 +1,63 @@
+package coop.rchain.p2p
+
+import coop.rchain.comm._, CommError._
+import scala.concurrent.duration.{Duration, MILLISECONDS}
+import coop.rchain.catscontrib._
+import cats._, cats.data._, cats.implicits._
+
+/** Eagerly evaluated instances to do reasoning about applied effects */
+object EffectsTestInstances {
+
+  class LogicalTime[F[_]: Capture] extends Time[F] {
+    var clock: Long = 0
+
+    def currentMillis: F[Long] = Capture[F].capture {
+      this.clock = clock + 1
```

Comment:
This races; not sure it matters, though.

Reply:
I guess, yeah. But this is just for tests which run in synchronous mode